vars/createReleaseTag.groovy

```diff
@@ -0,0 +1,35 @@
+def call(Map args = [:]) {
+
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def buildManifestObj = lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifest))
+
+    def componentsName = buildManifestObj.getComponets()
+    def componetsNumber = componentsName.size()
+    def version = args.tagVersion
+    echo "Creating $version release tag for $componetsNumber components in ths manifest"
+
+    withCredentials([usernamePassword(credentialsId: "${GITHUB_BOT_TOKEN_NAME}", usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_TOKEN')]) {
+        for (component in componentsName) {
```

Comment:
I think its better to add try catch and check if the tag already exists. Some components cut the tag beforehand.

Reply:
Agreed and probably can also check whether the existing tag has the correct commitid.

Reply:
I added a check below: If certain tag is existing in the current repo, we will just simply delete it and replace with the new one with the commit ID from the manifest.

Reply:
Not sure if we should delete or skip it. @dblock any thoughts on this?

Reply:
I would skip it with a note if it's identical, and fail if the existing tag doesn't match.

Reply:
I added another `if` block to check if the commit ID matches the ref for existing tag.

Reply:
Wouldn't it be easier using a try catch block so that exceptions can be handled per the scenario (Already existing and wrong ones). The code will be more cleaner too.

Reply:
I don't think there is try catch in shell. If we use the try catch outside of the shell block, push existing tag with same commit ID won't throw any exceptions. Error only occurs when we try to push tag with different commit ID.

Reply:
Did you try the groovy script or python for the same? Instead of inline code in JenkinsFile we can create a python script or groovy script and just execute it via jenkins as we are doing for other libraries. WDYT?

Reply:
This is the groovy library script that I'm trying to perform all functions and be executed from the Jenkins.

Reply:
Let me be more clear. I was recommending to remove the entire shell block and replace it with groovy. 
Just an example: https://stackoverflow.com/questions/45956629/how-to-use-git-commands-diff-log-etc-in-groovy-script-windows/45965983
Python: https://github.com/opensearch-project/opensearch-build/blob/main/src/git/git_repository.py

Reply:
Thanks @gaiksaya  for bringing this up. I too agree that we should remove the shell block, as much as we can, it is better for error handling and testing etc. 

@dblock - what are your thoughts here?

Reply:
@zelinh Can you work with the team to come up with an effort estimate to move the shell code with groovy? I would prefer to go that route if its a quick effort.

Reply:
I dislike the implementation here, but with the rule of thumb that this is better than what we had (manual process) I vote to merge this as is.

Reply:
Changed the implementation here and remove shell block as much as possible.

Reply:
> I would skip it with a note if it's identical (someone did the right thing and manually created the tag, it's ok), and fail if the existing tag doesn't match (someone created the wrong tag, let's find out loudly).

Hi @dblock What do you think about hard fail at the end rather than failing the entire workflow before all components are parsed? 
Right now this just fails the workflow and until that component fixes the tag we cannot re-run. How about go and finish the workflow and hard fail at the end if there is a mismatch?

Reply:
More elaboration on the current status: when some component already cut the tag with different commit from the commit in the distribution manifest, the current workflow will fail immediately and ask user to fix it until it matches.